src/Raven.Server/Documents/Includes/IncludeTimeSeriesCommand.cs

```diff
@@ -89,14 +93,27 @@ private List<TimeSeriesRangeResult> GetTimeSeriesForDocument(string docId, HashS
             return rangeResults;
         }
 
-        private List<TimeSeriesRangeResult> GetTimeSeriesForDocument(string docId, (IList<string> TimeseriesNames, IList<string> FromList, IList<string> ToList) timesSeriesToGet)
+        private List<TimeSeriesRangeResult> GetTimeSeriesForDocument(string docId, (IList<string> TimeSeriesNames, IList<string> FromList, IList<string> ToList) timesSeriesToGet)
         {
+            var stringsToDates = new Dictionary<string, DateTime>();
```

Comment:
I think it would be better to pass to this method a class/structure that has this cache so we can avoid parsing it per each document